File: js/ai.js
function  AI(computer, human)
{
    this.computer = computer;
	this.human = human;

    var that = this;

    this.addUnit = function() {
        this.computer.addUnit(["torso1"], 0xff0000);
    };

    this.update = function() {
        if (that.computer.energy > 5) that.addUnit(); // TODO

		for(var i = 0; i < that.computer.units.length; i++) {
			var compUnit = that.computer.units[i];
			var humUnit = that.human.units[0];

			if(!compUnit.isMoving() && humUnit) {
				var pos = humUnit.body.position.clone();
				pos.x += 1.5;
				compUnit.goTo(pos);
			}
		}
    };

    this.infoTimer = window.setInterval(this.update, 2000);
}

File: JME/src/cyberworld/AI.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package cyberworld;

import com.jme3.math.Vector3f;

/**
 *
 * @author demoniac
 */
public class AI {
    private Player computer;
    private Player human;
    private float targetCloseEnough = 4f;
    
    public AI(Player computer, Player human) {
        this.computer = computer;
        this.human = human;
    }
    
    private void addUnit() {
        computer.addUnit(new String[]{"torso1", "chassis1", "gun1"});
    };

    public void update(float tpf) {
        if(computer.getEnergy() > 7) addUnit(); // TODO

        if(human.getUnits().isEmpty()) return;

	for(Unit compUnit: computer.getUnits()) {
            Unit humUnitTarget = human.getUnits().get(0);
            float dist = Float.POSITIVE_INFINITY;
            for(Unit humUnit: human.getUnits()) {
                Vector3f humPos = humUnit.getPos();
                Vector3f compPos = compUnit.getPos();
                float distNew = compPos.distanceSquared(humPos);
                if(distNew < dist) {
                    dist = distNew;
                    humUnitTarget = humUnit;
                }
            }
            
            if(!compUnit.isMoving() && dist > targetCloseEnough*targetCloseEnough) {
                compUnit.setCloseEnough(targetCloseEnough);
                compUnit.goTo(humUnitTarget.getPos());
            }
	}
    };

    
}
